Travis is red though...

```
2022-02-10 12:25:50,299 7268 INFO openerp_test odoo.addons.account_asset_management.tests.test_account_asset_management: Starting TestAssetManagement.test_01_nonprorata_basic ...

2022-02-10 12:25:50,797 7268 INFO openerp_test odoo.addons.account_asset_management.tests.test_account_asset_management: ======================================================================

2022-02-10 12:25:50,797 7268 ERROR openerp_test odoo.addons.account_asset_management.tests.test_account_asset_management: ERROR: TestAssetManagement.test_01_nonprorata_basic

Traceback (most recent call last):

File "/home/travis/build/OCA/account-financial-tools/setup/account_asset_management/odoo/addons/account_asset_management/tests/test_account_asset_management.py", line 230, in test_01_nonprorata_basic

ict0.depreciation_line_ids[1].create_move()

File "/home/travis/build/OCA/account-financial-tools/setup/account_asset_management/odoo/addons/account_asset_management/models/account_asset_line.py", line 277, in create_move

move.action_post()

File "/home/travis/build/OCA/account-financial-tools/setup/account_asset_management/odoo/addons/account_asset_management/models/account_move.py", line 105, in action_post

aml.with_context(allow_asset=True).asset_id = asset.id

File "/home/travis/odoo-14.0/odoo/fields.py", line 1129, in __set__

records.write({self.name: write_value})

File "/home/travis/build/OCA/account-financial-tools/setup/account_asset_management/odoo/addons/account_asset_management/models/account_move.py", line 211, in write

"You cannot change an accounting item "

odoo.exceptions.UserError: You cannot change an accounting item linked to an asset depreciation line.
```

--- a/account_asset_management/models/account_move.py
+++ b/account_asset_management/models/account_move.py
@@ -93,6 +93,8 @@ class AccountMove(models.Model):
                     raise UserError(
                         _("Asset name must be set in the label of the line.")
                     )
+                if aml.asset_id:
+                    continue
                 asset_form = Form(
                     self.env["account.asset"]
                     .with_company(move.company_id)
@@ -102,7 +104,9 @@ class AccountMove(models.Model):
                     setattr(asset_form, key, val)
                 asset = asset_form.save()
                 asset.analytic_tag_ids = aml.analytic_tag_ids
-                aml.with_context(allow_asset=True).asset_id = asset.id
+                aml.with_context(
+                    allow_asset=True, allow_asset_removal=True
+                ).asset_id = asset.id
             refs = [
                 "<a href=# data-oe-model=account.asset data-oe-id=%s>%s</a>"
                 % tuple(name_get)
@@ -157,6 +161,9 @@ class AccountMoveLine(models.Model):
     asset_profile_id = fields.Many2one(
         comodel_name="account.asset.profile",
         string="Asset Profile",
+        compute="_compute_asset_profile",
+        store=True,
+        readonly=False,
     )
     asset_id = fields.Many2one(
         comodel_name="account.asset",
@@ -164,11 +171,13 @@ class AccountMoveLine(models.Model):
         ondelete="restrict",
     )
 
-    @api.onchange("account_id")
-    def _onchange_account_id(self):
-        if self.account_id.asset_profile_id:
-            self.asset_profile_id = self.account_id.asset_profile_id
-        super()._onchange_account_id()
+    @api.depends("account_id", "asset_id")
+    def _compute_asset_profile(self):
+        for rec in self:
+            if rec.account_id.asset_profile_id and not rec.asset_id:
+                rec.asset_profile_id = rec.account_id.asset_profile_id
+            elif rec.asset_id:
+                rec.asset_profile_id = rec.asset_id.profile_id
 
     @api.onchange("asset_profile_id")
     def _onchange_asset_profile_id(self):

